lua_modules/mcp23017/mcp23017.lua

```diff
@@ -0,0 +1,289 @@
+local MCP23017_IODIRB = 0x01
+local MCP23017_DEFVALA = 0x06
+local MCP23017_DEFVALB = 0x07
+local MCP23017_GPIOA = 0x12
+local MCP23017_GPIOB = 0x13
+
+--[[
+local MCP23017_IPOLA = 0x02
+local MCP23017_IPOLB = 0x03
+local MCP23017_GPINTENA = 0x04
+local MCP23017_GPINTENB = 0x05
+local MCP23017_DEFVALA = 0x06
+local MCP23017_DEFVALB = 0x07
+local MCP23017_INTCONA = 0x08
+local MCP23017_INTCONB = 0x09
+local MCP23017_IOCON = 0x0A
+local MCP23017_IOCON2 = 0x0B
+local MCP23017_GPPUA = 0x0C
+local MCP23017_GPPUB = 0x0D
+local MCP23017_INTFA = 0x0E
+local MCP23017_INTFB = 0x0F
+
+local MCP23017_INTCAPA = 0x10
+local MCP23017_INTCAPB = 0x11
+local MCP23017_OLATA = 0x14
+local MCP23017_OLATB = 0x15
+]]
+
+-- check device is available on address
+function mcp23017:checkDevice()
+    i2c.start(self.i2cId)
+    local response = i2c.address(self.i2cId, self.address, i2c.TRANSMITTER)
+    i2c.stop(self.i2cId)
+    if response ~= true then
+        print("MCP23017 device on " .. string.format('0x%02X', self.address) .. " not found")
+    end
+    return response
+end
+
+-- check device address (0x20 to 0x27)
+function mcp23017:checkAddress()
+    local addr = tonumber(self.address)
+    if (addr > 31 and addr < 40) then
+        return true
+    else
+        print("MCP23017 address is out of range")
+        return false
+    end
+end
+
+-- write byte
+function mcp23017:writeByte(registerAddr, val)
+    i2c.start(self.i2cId)
+    i2c.address(self.i2cId, self.address, i2c.TRANSMITTER)
+    i2c.write(self.i2cId, registerAddr)
+    i2c.write(self.i2cId, val)
+    i2c.stop(self.i2cId)
+end
+
+-- read byte
+function mcp23017:readByte(registerAddr)
+    i2c.start(self.i2cId)
+    i2c.address(self.i2cId, self.address, i2c.TRANSMITTER)
+    i2c.write(self.i2cId, registerAddr)
+    i2c.stop(self.i2cId)
+    i2c.start(self.i2cId)
+    i2c.address(self.i2cId, self.address, i2c.RECEIVER)
+    local data = i2c.read(self.i2cId, 1)
+    i2c.stop(self.i2cId)
+    return string.byte(data)
+end
+
+-- get IO dir register
+function mcp23017:getDirRegisterAddr(bReg)
+    if bReg == self.GPB then
+        return MCP23017_IODIRB
+    else
+        return MCP23017_IODIRA
+    end
+end
+
+-- get GPIO register address
+function mcp23017:getGPIORegisterAddr(bReg)
+    if bReg == self.GPB then
+        return MCP23017_GPIOB
+    else
+        return MCP23017_GPIOA
+    end
+end
+
+-- check pin is in range
+function mcp23017:checkPinIsInRange(pin)
+    if pin > 7 or pin < 0 then
+        print("The pin must be between 0 and 7")
+        return nil
+    end
+    return pin
+end
+
+-- setup internal pullup
+function mcp23017:setInternalPullUp(bReg, iByte)
+    if bReg == self.GPB then
+        self:writeByte(MCP23017_DEFVALB, iByte)
+    else
+        self:writeByte(MCP23017_DEFVALA, iByte)
+    end
+end
+
+-- set default GPIO mode
+function mcp23017:setDefaultMode(bReg, iByte)
+    self:writeByte(self:getDirRegisterAddr(bReg), iByte)
+end
+
+function mcp23017:numberToBool(val)
```

Comment:
This function seems unnecessary as only values like  `OUTPUT`, `INPUT`, `HIGH` and `LOW` (so only booleans) are allowed according to module documentation.

Reply:
This function was removed.